.github/workflows/binary-validator.yml

```
@@ -0,0 +1,18 @@
+name: Binary-validation
+# only run on PRs and when push a commit on a branch that we don't deploy on. 
+on: [push, pull_request]
+
+jobs:
+  api_check:
+    name: API check
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up JDK
+      uses: actions/setup-java@v2
+      with:
+        distribution: 'adopt'
+        java-version: '11'
+    - name: API check
+      run: ./gradlew apiCheck
```

If this fails, it might be difficult for the PR author to understand what happened and how to fix it themselves. 

What about adding a comment to the PR giving some help? 

```suggestion
    - name: API check
      run: ./gradlew apiCheck
      
    - uses: peter-evans/create-or-update-comment@v2
      if: ${{ failure() }} # only run if apiCheck fails. 
      with:
        token: ${{ secrets.READ_ONLY_BOT_TOKEN }}
        issue-number: ${{ github.event.pull_request.number }}
        body: Looks like the public facing API of the code base has changed. This could introduce bugs for customers if we are not careful. Please, check out [this doc](https://github.com/customerio/mobile/link/to/doc/on/binary/validator) to learn about this error and how to fix it. 
```